mono/mini/mini-llvm.c

```diff
@@ -4102,6 +4102,11 @@ process_call (EmitContext *ctx, MonoBasicBlock *bb, LLVMBuilderRef *builder_ref,
 	if (call->method && call->method->wrapper_type == MONO_WRAPPER_ALLOC)
 		mono_llvm_set_call_noalias_ret (lcall);
 
+	if (call->method && call->method->klass == mono_get_string_class () && !strcmp(call->method->name, "FastAllocateString")) {
```

Comment:
Hmm not very keep on hardcoding this in runtime (please add at least comment)

Do you have any perf numbers backing this?

Reply:
Better alternative would be attributes on the function?

Reply:
yeah I was thinking of special attributes for LLVM engine but I suspect CoreCLR/FX folks won't be happy about that. @marek-safar it removes a lot of redundant null checks (nonnull), Not sure about nonalias but I suspect it should allow LLVM to apply some vectorization or/and use memcpy instead of memmove (on *nix).

Reply:
`FastAllocateString` has 50 usages inside corelib (String, StringBuilder. Convert, Guid, NumberFormatting, ParseNumber, Enum, DateTimeFormat)

Reply:
We could for CoreLib recognize (C# non-nullable reference types hint) and apply nonnull everywhere but not sure how impactful is `nonalias` here

Reply:
cc @vargaz

Reply:
How about "LLVM sets the precedent"?

A lot of thought and design goes into LLVM.

They have these attributes.

There is a relationship between .NET and LLVM.
 For example LLVM being used to JIT or AOT compile .NET.

.NET should have these attributes too.